repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputeClasspath.java

```diff
@@ -111,6 +109,11 @@ public void testComputeClasspath() throws IOException {
         expectedClasspath.add(junit);
         expectedClasspath.add(hamcrest);
 
-        assertThat(jobStatus.getRepairClassPath(), is(expectedClasspath));
+        List<URL> actualClasspath = new ArrayList<>();
+        actualClasspath.add(classDir);
+        actualClasspath.add(testDir);
```

Comment:
This is not what I had in mine. I was thinking about creating manually the directories:

```
File classDir = new File(repoDir.getAbsolutePath(), "target/classes");
classDir.mkdir();

File testDir = new File(repoDir.getAbsolutePath(), "target/classes");
testDir.mkdir();
```

And then to check the result is good.

Reply:
Could you tell me why this way is better?

Supposing I agree with you, so what you are saying to me is to "break" the execution of the steps in the middle to create the directories?
It's like this now:
```
cloneStep.setNextStep(new CheckoutBuggyBuild(inspector, true))
                 .setNextStep(computeClasspath);
         cloneStep.execute();
```
So what you want is this:
```
cloneStep.setNextStep(new CheckoutBuggyBuild(inspector, true));
         cloneStep.execute();

// create directories

computeClasspath.execute();
```
?

Reply:
It don't think it's needed to break the execution. The way the test is built, the project is cloned exactly at path repoDir, so it should be ok to create before the cloning step the directories for repoDir/target/classes and test-classes.

Reply:
You didn't answer my first question, but I gave a shot for this even without understanding why I should do in your way, and it didn't work:

`11:56:03.323 [main] ERROR f.i.s.r.process.step.CloneRepository - Destination path "repo" already exists and is not an empty directory`

Reply:
> Could you tell me why this way is better?

I check the contract that `target/classes` and `target/test-classes` are added in the classpath. Here you're changing the output of the program to make it acceptable regarding the expected: it would be even better to change directly the expected to remove `classes` and `test-classes`.

Yeah I kind of expected this error to come up. Remove this check in CloneRepository: only warn the user. It should never happen for us as we are using it in a docker container.

Reply:
> I check the contract that target/classes and target/test-classes are added in the classpath. Here you're changing the output of the program to make it acceptable regarding the expected: it would be even better to change directly the expected to remove classes and test-classes.

Thanks for clarifying.